### ISSUE TITLE
Git push will fail when a trigger starts the build

### ISSUE THREAD

**Issue by fredrikhaglund** [Itelios/bitrise-steps-git-tag#6]
Bitrise git-clone will do git checkout with a commit hash so git push in this step will not work when a build is triggered form a pull request or commit. It only works when you start a build manually.

git push will give the following error when a bitrise build is triggered with a webhook from github.

```
git push --follow-tags
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD) state now, use 

    git push origin HEAD:<name-of-remote-branch>
```

**Comment by mseijas** [comment]
@fredrikhaglund have you found any workarounds for this? the same thing is happening to me

**Comment by fredrikhaglund** [comment]
This plug-in will never work for several reasons so I suggest that you use an ordinary script block instead. It is just one line of bash.

Before you can do any push from bitrise CI to git you need to manually create are read/write ssh-key instad of the read-only version created by bitrises built-in system.

If you use triggers to start your build you will do a git checkout of a specific commit so it is headless branch and that makes it harder to push changes back to git, too.

Note that bitrise always stores the git commit hash for each build so you can always find what you built without tags.

**Comment by vincentsaluzzo** [comment]
Resolved in commit 2909d9d
